api/blueprints/drbCollection.py

```diff
@@ -89,6 +92,74 @@ def collectionCreate(user=None):
 
     return APIUtils.formatOPDS2Object(201, opdsFeed)
 
+@collection.route('/update/<uuid>', methods=['POST'])
+@validateToken
+def collectionUpdate(uuid, user=None):
+    logger.info('Handling collection update request')
+
+    collectionData = request.json
+    dataKeys = collectionData.keys()
+
+    if {'title', 'creator', 'description'}.issubset(set(dataKeys)) == False\
+            or len(set(dataKeys) & set(['workUUIDs', 'editionIDs'])) == 0:
+        errMsg = {
+            'message':
+                'title, creator and description fields are required'
+                ', with one of workUUIDs or editionIDs to create a collection'
+        }
+
+        return APIUtils.formatResponseObject(400, 'createCollection', errMsg)
+
+    dbClient = DBClient(current_app.config['DB_CLIENT'])
+    dbClient.createSession()
+
+    #Getting the collection the user wants to replace
+    try:
+        collection = dbClient.fetchSingleCollection(uuid)
+    except NoResultFound:
+        errMsg = {'message': 'Unable to locate collection {}'.format(uuid)}
+        return APIUtils.formatResponseObject(404, 'fetchSingleCollection', errMsg)
+
+    workUUIDs=collectionData.get('workUUIDs', [])
+    editionIDs=collectionData.get('editionIDs', [])
+
+    collection.title = collectionData['title']
+    collection.creator = collectionData['creator']
+    collection.description = collectionData['description']
+
+    #Deleting the rows of collection_editions that were in the original collection
+    dbClient.session.execute(COLLECTION_EDITIONS.delete().where(COLLECTION_EDITIONS.c.collection_id == collection.id))
+
+    #Inserting rows of collection_editions based on workUUIDs/editionIDs arrays
+    if len(editionIDs) > 0:
+        for editID in editionIDs:
+            dbClient.session.execute(COLLECTION_EDITIONS.insert().values \
+                (collection_id = collection.id, edition_id = editID))
+
+    if len(workUUIDs) > 0:
+        collectionWorks = dbClient.session.query(Work)\
+            .join(Work.editions)\
+            .filter(Work.uuid.in_(workUUIDs))\
+            .all()
+
+        for work in collectionWorks:
+            editions = list(sorted(
+                    [ed for ed in work.editions],
+                    key=lambda x: x.publication_date
+                    if x.publication_date else date.today()
+            ))
+        
+            for edition in editions:
+                dbClient.session.execute(COLLECTION_EDITIONS.insert().values \
+                    (collection_id = collection.id, edition_id = edition.id))
```

Comment:
similar here, once you get all the edition ids from the works, you should just be able to insert in one go, instead of inserting one at a time in a loop. Also, is there a need to sort them?  It shouldn't actually affect how they're stored in the DB, but lmk if I'm missing something!

Reply:
You're right on the sorted method. I got lines 146-149 from another method and didn't realize that the sorted method isn't needed for this function in particular.